detekt-rules/src/main/kotlin/OrtImportOrder.kt

```diff
@@ -38,24 +40,93 @@ class OrtImportOrder : Rule() {
     override fun visitImportList(importList: KtImportList) {
         super.visitImportList(importList)
 
-        val importPaths = importList.imports.mapTo(mutableListOf()) { it.importPath.toString() }
-        val sortedImportPaths = importPaths.sorted().toMutableList()
-
-        // TODO: Also check for blank lines between imports from different top-level domains.
-        if (importPaths != sortedImportPaths) {
-            val path = importList.containingKtFile.absolutePath()
-            val message = "Imports in file '$path' are not sorted alphabetically"
-            val finding = CodeSmell(
-                issue,
-                // Use the message as the name to also see it in CLI output and not only in the report files.
-                Entity.from(importList).copy(name = message),
-                message
-            )
-            report(finding)
+        val children = importList.node.getChildren(null)
+        if (children.isNotEmpty()) {
+            val importPaths = children.mapNotNull {
+                when (it.psi) {
+                    is KtImportDirective -> (it.psi as KtImportDirective).importPath.toString()
+                    is PsiWhiteSpace -> if (it.textLength > 1) "" else null
+                    else -> null
+                }
+            }
+
+            val importPathsWithoutBlankLines = importPaths.filter(String::isNotEmpty).toMutableList()
+
+            val expectedImportOrder = createExpectedImportOrder(importPathsWithoutBlankLines)
+
+            if (importPaths != expectedImportOrder) {
+                val path = importList.containingKtFile.absolutePath()
+                val message = "Imports in file '$path' are not sorted alphabetically"
+                val finding = CodeSmell(
+                    issue,
+                    // Use the message as the name to also see it in CLI output and not only in the report files.
+                    Entity.from(importList).copy(name = message),
+                    message
+                )
+                report(finding)
+            }
+        }
+    }
+
+    private fun createExpectedImportOrder(importPaths: List<String>): List<String> {
+        val importPathsWithDot = mutableListOf<String>()
+        val importPathsWithoutDot = mutableListOf<String>()
+        val expectedImportPaths = mutableListOf<String>()
+
+        importPaths.forEach {
+            if (it.indexOf(".") == -1) {
+                importPathsWithoutDot.add(it)
+            } else {
+                importPathsWithDot.add(it)
+            }
+        }
+
+        importPathsWithoutDot.sort()
+        importPathsWithDot.sort()
+        val sortedImportPathsWithDotAndBlankLines = createImportListWithBlankLines(importPathsWithDot)
+
+        if (importPathsWithoutDot.isNotEmpty()) {
+            expectedImportPaths.addAll(importPathsWithoutDot)
+            expectedImportPaths.add("")
+        }
+
+        expectedImportPaths.addAll(sortedImportPathsWithDotAndBlankLines)
```

Comment:
For readability, please use the `+=` operator instead of `.addAll()` / `.add()`.